src-gui/ui/src/app/modules/switches/rule-details/rule-details.component.ts

```diff
@@ -25,8 +26,14 @@ export class RuleDetailsComponent implements OnInit {
     private toastr: ToastrService,
     private loaderService: LoaderService,
     private clipboardService: ClipboardService,
+    private router:Router,
     public commonService: CommonService
-  ) {}
+  ) {
+    if(!this.commonService.hasPermission('menu_switches')){
+      this.toastr.error('You are not authorised to access this page.');  

```

Comment:
Remove such messages from code.
Add these error messages in the properties file.

Reply:
added common message constant file